### ISSUE TITLE
Storyshots Puppeteer doesn't preserve authentication in Storybook URL

### ISSUE THREAD

**Issue by avendiart** [storybookjs/storybook#14107]
**Describe the bug**
Storyshots Puppeteer doesn't preserve authentication in Storybook URL

**To Reproduce**
Configure storybookUrl to include basic authentication in the URL, authentication is being dropped.
```typescript
const storyshotsConfig = {
  storybookUrl: 'http://username:password@example.com',
}
```

**Expected behavior**
Authentication information is preserved.

**Code snippets**
https://github.com/storybookjs/storybook/blob/79ed5f245949622cbd8ac2e3ba31a9e3ef409467/addons/storyshots/storyshots-puppeteer/src/url.ts#L8

**Comment by avendiart** [comment]
Resolved in #14582

**Comment by shilman** [comment]
Huzzah!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.3.0-alpha.8 containing PR #14582 that references this issue. Upgrade today to the `@next` NPM tag to try it out!

```
npx sb upgrade --prerelease
```
